### PR TITLE
Support different diagnostic level in tap file.

### PR DESCRIPTION
**Pull request by sophia-guo**
Support different diagnostic level in tap file. User can decide what
diagnostic information is needed. Values for DIAGNOSTICLEVEL are 'failure', 'all', 'nodetails'.
Also removed unused top level make variables

[ci skip]

Signed-off-by: Sophia Guo <sophiag@ca.ibm.com>

### CODE REVIEW

**Comment by smlambert** [comment]
What are the other possible values that one may set DIAGNOSTICLEVEL at?  
Is there a related update to the README?  
Is this to address #1112 ?

**Comment by sophia-guo** [comment]
No, this doesn't address #1112. Will update README.

**Comment by smlambert** [comment]
Thank you for the update, I now believe I know what this change is for.  :)

For background to others, the default setting of "failure" allows us to put a little extra detail into the otherwise extremely terse TAP (Test Anything Protocol - https://testanything.org/) output (its default is a 'no details' approach).  We grab the details of the test failure (stack trace and/or console output) and present it as part of the TAP result.  

We decided to use TAP as the "lowest common denominator / LCD" format for test output, because we run several different types of tests using different test frameworks, some of which do not use a standard format for their output.  By generating TAP output, we enforce an LCD so that we can present the results of all these varied tests and test frameworks under one common approach.

Your change allows a user to switch from the default setting of 'failure' to one of the other 2 choices by  exporting DIAGNOSTICLEVEL, if they want more or less detail produced.  This may be desirable, if someone is running tests locally and suddenly have 100s of failures, and want to rerun, but not scroll through reams of failure details (especially if they are all failing for the same reason and developer is trial'n'erroring some changes for a fix).

File: test/README.md [hunk 1]
```diff
@@ -281,7 +281,21 @@ disabled.spec.<spec> (e.g. disabled.spec.linux_x86-64)
     or were skipped, along with other information like execution time and
     error messages, exceptions and logs from the individual test methods.
 
+  * tap result files
```
Comment (smlambert) [inline test/README.md]: Please change to TAP (Test Anything Protocol) result files  (and change any subsequent use of tap to TAP as its an acronym).

**Comment by sophia-guo** [comment]
@smlambert Thanks! Updated tap to TAP.

**Pull request merged**